sql/src/main/java/io/crate/expression/reference/file/LineContext.java

```diff
@@ -66,8 +58,27 @@ public Object get(ColumnIdent columnIdent) {
         return val;
     }
 
+    public Map<String, Object> sourceAsMap() {
+        if (parsedSource == null) {
+            try {
+                parsedSource = XContentHelper.convertToMap(new BytesArray(rawSource), false, XContentType.JSON).v2();
+            } catch (NullPointerException e) {
+                return null;
+            }
+        }
+        return parsedSource;
+    }
+
     public void rawSource(byte[] bytes) {
         this.rawSource = bytes;
         this.parsedSource = null;
     }
+
+    public void rawSourceFromCSV(byte[] header, byte[] line) throws IOException {
+        CSVLineParser csvParser = new CSVLineParser();
```

Comment:
Since the Parser doesn't carry any state it would be possible to make the `parse` method static and avoid the instance creation

Reply:
Good point 👍